core/src/main/java/com/datastrato/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -102,4 +110,61 @@ public static void checkRoleNamespace(Namespace namespace) {
         "Role namespace must have 3 levels, the input namespace is %s",
         namespace);
   }
+
+  public static void checkPermission(String metalake, SecurableObject object) {
+
+    if (!satisfyPrivileges(metalake, object)) {
+      throw new ForbiddenException(
+          "Securable object %s doesn't have the one of privileges %s",
+          object,
+          StringUtils.join(
+              object.privileges().stream().map(Privilege::name).collect(Collectors.toList()), ","));
+    }
+  }
+
+  public static boolean satisfyPrivileges(String metalake, SecurableObject object) {
+    if (object.privileges().isEmpty()) {
+      return true;
+    }
+
+    String currentUser = PrincipalUtils.getCurrentUserName();
+
+    AccessControlManager accessControlManager = GravitinoEnv.getInstance().accessControlManager();
+    List<RoleEntity> roles;
+    try {
+      roles = accessControlManager.getRolesByUserFromMetalake(metalake, currentUser);
+
+      for (RoleEntity role : roles) {
+        for (Privilege privilege : object.privileges()) {
+          if (role.hasPrivilegeWithCondition(object, privilege.name(), Privilege.Condition.DENY)) {
+            continue;
+          }
+
+          if (role.hasPrivilegeWithCondition(object, privilege.name(), Privilege.Condition.ALLOW)) {
+            return true;
+          }
```

Comment:
I don't understand this block, assuming a role has this privileges: "deny on metalake about create table" and the `SecurableObject` is "deny on a schema of the metalake about create table", In my opinion, `satisfyPrivileges` should return true.

Reply:
I will add some comments for this.